content/tutorial-deep-learning-on-mnist.ipynb

```diff
@@ -0,0 +1,1017 @@
+{
```

Comment:
<p>Can you add a sentence about why you are initializing the weights like this? Is that standard practice in machine learning, or is it really the choice of the user?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/numpy/numpy-tutorials/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='65'/>

Reply:
Absolutely @melissawm . I also added a note on "Xavier" initialization in the other tutorial - https://www.deeplearning.ai/ai-notes/initialization/ (it's also a very useful link for the reader).

Reply:
A very good question. I added this to the explainer section. It's at the heart of the algorithm, with it being a stochastic method.

```diff
 - _Weights_: These are important adjustable parameters that the neural network 
 fine-tunes by forward and backward propagating the data. 
+ They are optimized through a process called 
+ [gradient descent](https://en.wikipedia.org/wiki/Stochastic_gradient_descent). 
+ Before the model training starts, the weights are randomly initialized with 
+ NumPy's `np.random.random()` function.
    
    The optimal weights should produce the highest prediction accuracy and the lowest error on the training and test sets. 
- Before the model training starts, the weights are randomly initialized with 
- NumPy's `np.random.random()` function.
```